src/System.Threading.Channels/src/System/Threading/Channels/Interactor.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+
+namespace System.Threading.Channels
+{
+    internal abstract class Interactor<T> : TaskCompletionSource<T>
```

Comment:
Might be worth adding some comments to explain what the interactors are for,

Reply:
Sure, will add comments. Will also try to come up with a better name.